core_backend/app/schemas.py

```diff
@@ -103,42 +110,82 @@ class FeedbackBase(BaseModel):
     model_config = ConfigDict(from_attributes=True)
 
 
-class ContentCreate(BaseModel):
+class LanguageBase(BaseModel):
+    """
+    Pydantic model for language
+    """
+
+    language_name: str
+    is_default: bool
+    model_config = ConfigDict(from_attributes=True)
+
+    @validator("language_name")
+    def language_name_must_be_valid(cls, value: str) -> str:
+        # Regex pattern allows letters, spaces, hyphens, and apostrophes
+        if not re.match(r"^[A-Za-zÀ-ÖØ-öø-ÿ\-' ]+$", value):
+            raise ValueError("Invalid characters in language name.")
+        return value
+
+
+class LanguageRetrieve(LanguageBase):
+    """
+    Pydantic model for language retrieval
+    """
+
+    language_id: int
+    created_datetime_utc: datetime
+    updated_datetime_utc: datetime
+
+    model_config = ConfigDict(from_attributes=True)
+
+
+class ContentTextCreate(BaseModel):
     """
     Pydantic model for content creation
     """
 
     # Ensure len("*{title}*\n\n{text}") <= 1600
     content_title: Annotated[str, StringConstraints(max_length=150)]
     content_text: Annotated[str, StringConstraints(max_length=2000)]
-    content_language: str = "ENGLISH"
+    language_id: int
+    content_id: Optional[int] = Field(
+        default=None,
+        description="If adding or editing content text to an existing content"
+        ", provide its ID",
+    )
+
     content_metadata: dict = {}
 
     model_config = ConfigDict(from_attributes=True)
 
-    @validator("content_language")
-    def validate_language(cls, v: str) -> str:
-        """
-        Validator for language
-        """
 
-        if v not in IdentifiedLanguage.get_supported_languages():
-            raise ValueError(f"Language {v} is not supported")
+class ContentRetrieve(ContentTextCreate):
+    """
+    Pydantic model for content retrieval
+    """
 
-        return v
+    content_text_id: int
+    created_datetime_utc: datetime
+    updated_datetime_utc: datetime
 
 
-class ContentRetrieve(ContentCreate):
+class ContentSummary(BaseModel):
     """
-    Pydantic model for content retrieval
+    Pydantic model
+    for content summary
     """
 
+    content_text_id: int
     content_id: int
+    content_title: str
+    languages: list[str]
```

Comment:
can we rename this to `available_languages`? Because the content summary endpoint requires a language_id, seeing languages returned was a bit confusing. Or add some JSON schema description?